smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/swiftintegrations/TestEquatableConformanceIntegration.kt

```diff
@@ -0,0 +1,86 @@
+package software.amazon.smithy.swift.codegen.swiftintegrations
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.model.shapes.DoubleShape
+import software.amazon.smithy.model.shapes.FloatShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.shapes.UnionShape
+import software.amazon.smithy.model.traits.ErrorTrait
+import software.amazon.smithy.swift.codegen.SwiftDelegator
+import software.amazon.smithy.swift.codegen.SwiftDependency
+import software.amazon.smithy.swift.codegen.core.SwiftCodegenContext
+import software.amazon.smithy.swift.codegen.customtraits.EquatableConformanceTrait
+import software.amazon.smithy.swift.codegen.customtraits.TestEquatableConformanceTrait
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.hasTrait
+import software.amazon.smithy.swift.codegen.swiftmodules.SwiftTypes
+
+class TestEquatableConformanceIntegration : SwiftIntegration {
```

Comment:
This SwiftIntegration does the code generation of Equatable conformance for types that need Equatable for protocol response tests, but don't already have it auto-rendered in the SDK.

The rendering code is copied straight out of the `HttpProtocolUnitTestResponseGenerator`.